hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/recovery/ReconOMMetadataManager.java

```diff
@@ -47,4 +50,26 @@ public interface ReconOMMetadataManager extends OMMetadataManager {
    * @return true if OM Tables are initialized, otherwise false.
    */
   boolean isOmTablesInitialized();
+
+  /**
+   * Return all volumes in the file system.
+   * @return all the volumes from the OM DB.
+   */
+  List<OmVolumeArgs> listVolumes() throws IOException;
```

Comment:
we should not load all volume list in memory. Please use pagination and index concept. Need to keep the system bounded.

Reply:
Updated with limit (default 1000) to bound the number buckets/volumes returned.

Reply:
> Updated with limit (default 1000) to bound the number buckets/volumes returned.

Refers to `ReconOMMetadataManager#listVolumes(String startKey, int maxKeys)` which will be called instead of `listVolumes()`.